src/nvim/message.c

```diff
@@ -3240,46 +3230,41 @@ int vim_vsnprintf(char *str, size_t str_m, char *fmt, va_list ap, typval_T *tvs)
         }
       }
 
-      /* parse 'h', 'l' and 'll' length modifiers */
-      if (*p == 'h' || *p == 'l') {
+      // parse 'h', 'l', 'll' and 'z' length modifiers
+      if (*p == 'h' || *p == 'l' || *p == 'z') {
         length_modifier = *p;
         p++;
-        if (length_modifier == 'l' && *p == 'l') { /* double l = long long */
-          length_modifier = '2';                   /* double l encoded as '2' */
+        if (length_modifier == 'l' && *p == 'l') { // ll, encoded as 2
+          length_modifier = '2';
           p++;
         }
       }
+
       fmt_spec = *p;
 
-      /* common synonyms: */
+      // common synonyms
       switch (fmt_spec) {
-      case 'i': fmt_spec = 'd'; break;
-      case 'D': fmt_spec = 'd'; length_modifier = 'l'; break;
-      case 'U': fmt_spec = 'u'; length_modifier = 'l'; break;
-      case 'O': fmt_spec = 'o'; length_modifier = 'l'; break;
-      case 'F': fmt_spec = 'f'; break;
-      default: break;
+        case 'i': fmt_spec = 'd'; break;
+        case 'D': fmt_spec = 'd'; length_modifier = 'l'; break;
+        case 'U': fmt_spec = 'u'; length_modifier = 'l'; break;
+        case 'O': fmt_spec = 'o'; length_modifier = 'l'; break;
+        case 'F': fmt_spec = 'f'; break;
+        default: break;
       }
 
-      /* get parameter value, do initial processing */
+      // get parameter value, do initial processing
       switch (fmt_spec) {
-      /* '%' and 'c' behave similar to 's' regarding flags and field widths */
-      case '%':
-      case 'c':
-      case 's':
-      case 'S':
+      // '%' and 'c' behave similar to 's' regarding flags and field widths
+      case '%': case 'c': case 's': case 'S':
```

Comment:
Personally, I like this. In some situations...
```c
case ENUM_VAL1: ...
case ENUM_VAL2: ...
...
```
I think that lining up the labels looks much clearer, but for cases like this...
```c
case 0: case 1: case 2: ...
```
where they're literal, it wastes less screen space.